openraft/src/core/admin.rs

```diff
@@ -204,6 +212,20 @@ impl<'a, C: RaftTypeConfig, N: RaftNetworkFactory<C>, S: RaftStorage<C>> LeaderS
 
         tracing::debug!(?new_config, "new_config");
 
+        if let Err(e) = self.is_all_new_members_line_rate(new_members.cloned().collect::<BTreeSet<_>>(), blocking) {
+            let _ = tx.send(Err(e));
+            return Ok(());
+        }
+
+        self.append_membership_log(new_config, Some(tx)).await?;
+        Ok(())
+    }
+
+    fn is_all_new_members_line_rate(
```

Comment:
This method is not just about `new members`:

```suggestion
    fn are_nodes_at_line_rate(
```